api/src/main/java/edu/cornell/mannlib/vitro/webapp/searchindex/extensions/VCardsAcrossContextNodes.java

```diff
@@ -0,0 +1,455 @@
+                        .execute().toStringFields("vName").flatten());
+    }
+
+    private void addVCardsFromContextNodeClass(Individual ind, SearchInputDocument doc, String contextNodeClass) {
+        addVCardsToTextFields(
+                doc,
+                createSelectQueryContext(rdfService,
+                        LABELS_FOR_SPECIFIC_CONTEXT_NODE_TYPE)
+                        .bindVariableToUri("uri", ind.getURI())
+                        .bindVariableToUri("nodeType", contextNodeClass)
+                        .bindVariableToUri("incoming", incomingPropertyUri)
+                        .bindVariableToUri("outgoing", outgoingPropertyUri)
+                        .execute().toStringFields("vName").flatten());
+    }
+
+    private void addVCardsToTextFields(SearchInputDocument doc, List<String> vcardUris) {
+        for (String  vcardUri : vcardUris) {
+            String value = vcardToString(vcardUri);
+            if (!StringUtils.isEmpty(value)) {
+                doc.addField(ALLTEXT, value);
+                doc.addField(ALLTEXTUNSTEMMED, value);
+            }
+        }
+    }
+
+    private String vcardToString(String vcardUri) {
+        String givenName = null;
+        String familyName = null;
+
+        String sparqlConstruct = "CONSTRUCT\n"
+                + "{\n"
+                + "    <" + vcardUri + "> ?p ?o .\n"
+                + "}\n"
+                + "WHERE\n"
+                + "{\n"
+                + "    <" + vcardUri + "> ?p ?o .\n"
+                + "}\n";
+
+        try {
+            Model model = ModelFactory.createDefaultModel();
+            rdfService.sparqlConstructQuery(sparqlConstruct, model);
+
+            StmtIterator iter = model.listStatements();
+            try {
+                while (iter.hasNext()) {
+                    Statement stmt = iter.nextStatement();
+
+                    if (stmt.getObject().isLiteral()) {
+                        switch (stmt.getPredicate().getURI()) {
+                            case "http://www.w3.org/2006/vcard/ns#familyName":
+                                familyName = stmt.getObject().asLiteral().getString();
+                                break;
+
+                            case "http://www.w3.org/2006/vcard/ns#givenName":
+                                givenName = stmt.getObject().asLiteral().getString();
+                                break;
+                        }
+                    }
+                }
+            } finally {
+                iter.close();
+            }
+        } catch (RDFServiceException e) {
+            log.error(e, e);
+        }
+
+        if (!StringUtils.isEmpty(familyName)) {
+            if (!StringUtils.isEmpty(givenName)) {
+                return familyName + ", " + givenName;
+            } else {
+                return familyName;
+            }
+        } else if (!StringUtils.isEmpty(givenName)) {
+            return givenName;
+        }
+
+        return null;
```

Comment:
Isn't it possible to avoid the conditioning above by constructing the return value in the iteration?